pkg/adapter/incoming.go

```diff
@@ -86,6 +112,78 @@ func (l *listener) detectIncoming(ctx context.Context, req *http.Request, payloa
 	return true, repo, nil
 }
 
+func (l *listener) getAndUpdateInstallationID(ctx context.Context, req *http.Request, run *params.Run, repo *v1alpha1.Repository) error {
+	jwtToken, err := GenerateJWT(ctx, run)
+	if err != nil {
+		return err
+	}
+
+	enterpriseURL := req.Header.Get("X-GitHub-Enterprise-Host")
+	if enterpriseURL != "" {
+		apiURL = enterpriseURL
+	}
+
+	res, err := getResponse(ctx, http.MethodGet, installationURL, jwtToken, run)
+	if err != nil {
+		return err
+	}
+
+	defer res.Body.Close()
+	data, err := io.ReadAll(res.Body)
+	if err != nil {
+		return err
+	}
+
+	installationData := []gt.Installation{}
+	if err = json.Unmarshal(data, &installationData); err != nil {
+		return err
+	}
+
+	/* each installationID can have list of repository
+	ref: https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps#authenticating-as-an-installation ,
+	     https://docs.github.com/en/rest/apps/installations?apiVersion=2022-11-28#list-repositories-accessible-to-the-app-installation */
+	for i := range installationData {
+		l.event.Provider.URL = enterpriseURL
```

Comment:
isn't that a bug ? and you don't have fallback in the previous logic to the public url

Reply:
fixed that 

```
l.event.Provider.URL = enterpriseURL
```

if `l.event.Provider.URL` is empty then it will use https://github.com/openshift-pipelines/pipelines-as-code/blob/main/pkg/provider/github/parse_payload.go#L61-L63 public one